upstream/upstream.go

```diff
@@ -66,3 +70,35 @@ const (
 
 	DefaultSemVerConstraints = ">= 0.0.0"
 )
+
+func selectHighestVersion(constraints string, expectedRange semver.Range, tags []string) (string, error) {
+	var candidateVersion semver.Version
+	candidateVersionString := "" // keep the version string separately as it may contain a leading `v`
+	for _, tag := range tags {
+		// Try to match semver and range
+		version, err := semver.Parse(strings.Trim(tag, "v"))
+		if err != nil {
+			log.Debugf("Error parsing version %s (%#v) as semver, cannot validate semver constraints", tag, err)
+			continue
+		}
+
+		if !expectedRange(version) {
+			log.Debugf("Skipping release not matching range constraints (%s): %s\n", constraints, tag)
```

Comment:
There is no need for the newline on logrus statements:

```suggestion
			log.Debugf("Skipping release not matching range constraints (%s): %s", constraints, tag)
```